zerver/webhooks/githubsponsors/tests.py

```diff
@@ -0,0 +1,79 @@
+from unittest.mock import patch
+
+from zerver.lib.test_classes import WebhookTestCase
+
+
+class GitHubSponsorsHookTests(WebhookTestCase):
+    STREAM_NAME = "github"
+    URL_TEMPLATE = "/api/v1/external/githubsponsors?stream={stream}&api_key={api_key}"
+    WEBHOOK_DIR_NAME = "githubsponsors"
+
+    def test_cancelled_message(self) -> None:
+        expected_topic = "sponsors"
+        expected_message = "monalisa cancelled their $5 a month subscription."
+        self.check_webhook(
+            "cancelled",
+            expected_topic,
+            expected_message,
+        )
+
+    def test_created_message(self) -> None:
+        expected_topic = "sponsors"
+        expected_message = "monalisa subscribed for $5 a month."
+        self.check_webhook(
+            "created",
+            expected_topic,
+            expected_message,
+        )
+
+    def test_pending_cancellation_message(self) -> None:
+        expected_topic = "sponsors"
+        expected_message = (
+            "monalisa's $5 a month subscription will be cancelled on January 05, 2020."
+        )
+        self.check_webhook(
+            "pending_cancellation",
+            expected_topic,
+            expected_message,
+        )
+
+    def test_pending_tier_change_message(self) -> None:
+        expected_topic = "sponsors"
+        expected_message = "monalisa's subscription will change from $10 a month to $5 a month on December 30, 2019."
+        self.check_webhook(
+            "pending_tier_change",
+            expected_topic,
+            expected_message,
+        )
+
+    def test_tier_changed_message(self) -> None:
+        expected_topic = "sponsors"
+        expected_message = "monalisa changed their subscription from $10 a month to $5 a month."
+        self.check_webhook(
+            "tier_changed",
+            expected_topic,
+            expected_message,
+        )
+
+    def test_ignored_events(self) -> None:
+        # The payload for these events never gets looked at in the
+        # webhook itself; it only needs to be valid JSON.
+        payload = "{}"
+
+        ignored_events = [
+            "edited",
+        ]
```

Comment:
Is there any particular reason the `edited` event is being ignored here?

Reply:
The main use case of this integration is monitoring new sign ups and changes in subscriptions, but as noted [here](https://docs.github.com/en/webhooks/webhook-events-and-payloads?actionType=edited#sponsorship) "the edited event types include the details about the change when someone edits a sponsorship to change the privacy". I figured it's probably not relevant for this use case, but maybe we should implement it anyway?